store/tikv/2pc.go

```diff
@@ -626,6 +685,10 @@ func (c *twoPhaseCommitter) pessimisticLockSingleBatch(bo *Backoffer, batch batc
 		lockResp := resp.Resp.(*pb.PessimisticLockResponse)
 		keyErrs := lockResp.GetErrors()
 		if len(keyErrs) == 0 {
+			isPrimary := bytes.Equal(batch.keys[0], c.primary())
+			if isPrimary { // No need to check isPessimistic because this function is only called in that case.
+				c.ttlManager.run(c)
```

Comment:
Maybe I change the txnHeartBeat proto to return the lock status, if the lock is committed/rollback, let the goroutine return?

Reply:
@coocood disagree with the change on kvproto @lysu 

I add `committer.ttlManager.close()` to Rollback, if a pessimistic txn doesn't run execute(), it would run Rollback to release the goroutine.